sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala

```diff
@@ -20,34 +21,45 @@ object AkkaGrpcPlugin extends AutoPlugin {
   override def requires = ProtocPlugin
 
   trait Keys { _: autoImport.type =>
+
+    object AkkaGrpc {
+      sealed trait TargetStub
+      case object Client extends TargetStub
+      case object Server extends TargetStub
+
+      sealed trait TargetLanguage
+      case object Scala extends TargetLanguage
+      case object Java extends TargetLanguage
+    }
+
+    val akkaGrpcTargetLanguages = settingKey[Seq[AkkaGrpc.TargetLanguage]]("Which languages to generate service and model classes for (AkkaGrpc.Scala, AkkaGrpc.Java)")
+    val akkaGrpcTargetStubs = settingKey[Seq[AkkaGrpc.TargetStub]]("Which of the stubs to generate (AkkaGrpc.Server, AkkaGrpc.Client)")
     val akkaGrpcCodeGeneratorSettings = settingKey[Seq[String]]("Settings to pass to the code generator")
-    val akkaGrpcCodeGenerators = settingKey[Seq[GeneratorAndSettings]]("The configured source generator")
-    val akkaGrpcModelGenerators = settingKey[Seq[Target]]("The configured source generator for model classes")
   }
 
-  object autoImport extends Keys {
-    final case class GeneratorAndSettings(generator: CodeGenerator, settings: Seq[String] = Nil)
-  }
+  object autoImport extends Keys
   import autoImport._
 
-  override def projectSettings = defaultSettings ++ configSettings(Compile) ++ configSettings(Test)
+  override def projectSettings: Seq[sbt.Setting[_]] = defaultSettings ++ configSettings(Compile) ++ configSettings(Test)
 
   private def defaultSettings =
     Seq(
       akkaGrpcCodeGeneratorSettings := Seq("flat_package"),
-      akkaGrpcCodeGenerators := GeneratorAndSettings(ScalaBothCodeGenerator, akkaGrpcCodeGeneratorSettings.value) :: Nil)
+      akkaGrpcTargetStubs := Seq(AkkaGrpc.Client, AkkaGrpc.Server),
+      akkaGrpcTargetLanguages := Seq(AkkaGrpc.Scala))
 
   def configSettings(config: Configuration): Seq[Setting[_]] =
     inConfig(config)(Seq(
-      akkaGrpcModelGenerators := Seq[Target]((JvmGenerator("scala", ScalaPbCodeGenerator), akkaGrpcCodeGeneratorSettings.value) -> sourceManaged.value),
-
       unmanagedResourceDirectories ++= (resourceDirectories in PB.recompile).value,
       watchSources in Defaults.ConfigGlobal ++= (sources in PB.recompile).value,
 
       // configure the proto gen automatically by adding our codegen:
+      // FIXME: actually specifying separate Compile and Test target stub and languages does not work
```

Comment:
Seems fine, perhaps let's add a ticket making it explicit that we currently don't support this?